core/main.go

```diff
@@ -16,21 +16,37 @@ func main() {
 	logger.Init(format, level)
 
 	logrus.Println("Starting MESG Core", version.Version)
-	go startServer(&api.Server{
+
+	tcpServer := &api.Server{
 		Network: "tcp",
 		Address: viper.GetString(config.APIServerAddress),
-	})
-	go startServer(&api.Server{
+	}
+
+	unixServer := &api.Server{
 		Network: "unix",
 		Address: viper.GetString(config.APIServerSocket),
-	})
+	}
+
+	go startServer(tcpServer)
+	go startServer(unixServer)
+
+	closing := make(chan struct{}, 2)
```

Comment:
I'm not sure I understand how this is useful.
Why not doing something like 
```go
defer tcpService.Close()
defer unixServer.Close()
```
instead of having to create this chan that we pass as a parameters of the `closeServer` we could close the server directly and not in some goroutines. I might be missing something here

Reply:
yes actually this is needed. Close() will stop accepting incoming gRPC requests and gracefully wait current ones to end. so Close() will block until all accepted gRPC requests are done. We should only exit the program after both Close() calls returned.

Reply:
and the reason why we use a goroutine and a chan is to close both server concurrently and don't force one to wait until another one closes.